biz.aQute.launcher/src/aQute/launcher/Launcher.java

```diff
@@ -754,8 +757,12 @@ private void doSecurity() {
 
 	@SuppressWarnings("deprecation")
 	private boolean isFragment(Bundle b) {
-		return padmin != null
-				&& padmin.getBundleType(b) == org.osgi.service.packageadmin.PackageAdmin.BUNDLE_TYPE_FRAGMENT;
+		if(padmin != null){
+			return padmin.getBundleType(b) == PackageAdmin.BUNDLE_TYPE_FRAGMENT;
+		} else {
+			BundleRevision rev = b.adapt(BundleRevision.class);
+			return !rev.getDeclaredRequirements("osgi.wiring.host").isEmpty();
```

Comment:
This is the wrong test. According to the Core spec (section 7.4), the test should be:

```
BundleRevision rev = aBundle.adapt(BundleRevision.class);
if ( rev != null && (rev.getTypes() & BundleRevision.TYPE_FRAGMENT)!= 0 ) {
     ... // do the fragment thing
}
```